sample-code/spring-app/src/test/java/com/sap/ai/sdk/app/services/OpenAiServiceV2.java

```diff
@@ -130,10 +124,8 @@ public OpenAiChatCompletionResponse chatCompletionTools(final int months) {
    * @param unit The unit of temperature to use.
    * @return The assistant message response.
    */
-  @Nonnull
-  public CreateChatCompletionResponse chatCompletionToolExecution(
+  public OpenAiChatCompletionResponse chatCompletionToolExecution(
       @Nonnull final String location, @Nonnull final String unit) {
-
     final var schemaMap =
         Try.of(() -> new JsonSchemaGenerator(JACKSON).generateSchema(WeatherMethod.Request.class))
             .map(schema -> JACKSON.convertValue(schema, Map.class))
```

Comment:
```suggestion
            .map(
                schema -> JACKSON.convertValue(schema, new TypeReference<Map<String, Object>>() {}))
```
This will remove the unchecked class cast warning